source/common/tracing/zipkin/tracer.cc

```diff
@@ -70,7 +76,7 @@ SpanPtr Tracer::startSpan(const std::string& span_name, SystemTime timestamp,
 
     // Set the timestamp globally for the span
     span_ptr->setTimestamp(timestamp_micro);
-  } else if (previous_context.annotationSet().cs_ && !previous_context.annotationSet().sr_) {
```

Comment:
Is it safe to disregard altogether the previous context when making this decision?

Reply:
As above.